tests/plugins/test_default_qubit_autograd.py

```diff
@@ -0,0 +1,341 @@
+            assert hasattr(circuit, "jacobian")
+
+        def cost(params):
+            """Perform some classical processing"""
+            return circuit(params[0], params[1:], w=0) ** 2
+
+        theta = 0.543
+        phi = -0.234
+        lam = 0.654
+
+        params = np.array([theta, phi, lam], requires_grad=True)
+
+        res = cost(params)
+        expected_cost = (np.sin(lam) * np.sin(phi) - np.cos(theta) * np.cos(lam) * np.cos(phi)) ** 2
+        assert np.allclose(res, expected_cost, atol=tol, rtol=0)
+
+        res = qml.grad(cost)(params)
+        expected_grad = (
+            np.array(
+                [
+                    np.sin(theta) * np.cos(lam) * np.cos(phi),
+                    np.cos(theta) * np.cos(lam) * np.sin(phi) + np.sin(lam) * np.cos(phi),
+                    np.cos(theta) * np.sin(lam) * np.cos(phi) + np.cos(lam) * np.sin(phi),
+                ]
+            )
+            * 2
+            * (np.sin(lam) * np.sin(phi) - np.cos(theta) * np.cos(lam) * np.cos(phi))
+        )
+        assert np.allclose(res, expected_grad, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("interface", ["tf", "torch"])
+    def test_error_backprop_wrong_interface(self, interface, tol):
+        """Tests that an error is raised if diff_method='backprop' but not using
+        the Autograd interface"""
+        dev = qml.device("default.qubit.autograd", wires=1)
+
+        def circuit(x, w=None):
+            qml.RZ(x, wires=w)
+            return qml.expval(qml.PauliX(w))
+
+        with pytest.raises(
+            ValueError,
+            match="default.qubit.autograd only supports diff_method='backprop' when using the autograd interface",
+        ):
+            qml.qnode(dev, diff_method="backprop", interface=interface)(circuit)
+
+
+class TestHighLevelIntegration:
+    """Tests for integration with higher level components of PennyLane."""
+
+    @pytest.mark.xfail(
+        strict=True,
+        reason="qml.broadcast currently performs input type validation, only allowing Nones or Iterables. "
+        "The Autograd Arraybox is not an iterable, so an error is raised. This can be avoided by "
+        "simply modifying this type validation.",
+    )
+    def test_qnode_collection_integration(self):
+        """Test that a PassthruQNode default.qubit.autograd works with QNodeCollections."""
+        dev = qml.device("default.qubit.autograd", wires=2)
+
+        obs_list = [qml.PauliX(0) @ qml.PauliY(1), qml.PauliZ(0), qml.PauliZ(0) @ qml.PauliZ(1)]
+        qnodes = qml.map(
+            qml.templates.StronglyEntanglingLayers, obs_list, dev, interface="autograd"
+        )
+
+        assert qnodes.interface == "autograd"
+
+        weights = np.array(
+            qml.init.strong_ent_layers_normal(n_wires=2, n_layers=2), requires_grad=True
+        )
+
+        def cost(weights):
+            return np.sum(qnodes(weights))
+
+        grad = qml.grad(cost)(weights)[0]
+        assert grad.shape == weights.shape

```

Comment:
I'd recommend adding unit tests for `_reduce_sum`

Reply:
Have removed custom `_reduce_sum` implementation :)